microsoft-edge/devtools-guide-chromium/memory-problems/memory-101.md

```diff
@@ -23,165 +23,146 @@ ms.date: 07/21/2023
+To learn more about using the **Memory** tool, see [Record heap snapshots using the Memory tool](./heap-snapshots.md).
 
 
 <!-- ====================================================================== -->
-## Object sizes
+## The memory graph
 
-Think of memory as a graph with primitive types (like numbers and strings) and objects (associative arrays).  Memory can be visually represented as a graph with a number of interconnected points, as follows:
+Think of the memory used by a webpage as a graph: a structure that contains nodes that are connected together by edges. The nodes in the memory graph represent the objects used by the page, such as primitive types like JavaScript numbers or strings, or objects like associative arrays.
 
 ![Visual representation of memory](./memory-101-images/thinkgraph.png)
 
-An object can hold memory in two ways:
+Nodes and edges in the memory graph are given the following labels:
 
-*  Directly; the memory is held by the object itself.
+*  _Nodes_ (or _objects_) are labelled using the name of the _constructor_ function that was used to build them.
 
-*  Implicitly, by holding references to other objects.  An object holding references to other objects prevents those objects from being automatically disposed by a garbage collector (GC).
+*  _Edges_ are labelled using the names of _properties_.
 
-The [Memory](heap-snapshots.md) panel in DevTools is a tool for investigating memory issues.
+A node in the graph can hold memory in two ways:
 
-When working with the Memory panel, you will likely find yourself looking at a few different columns of information.  Two columns that stand out are **Shallow Size** and **Retained Size**:
+*  Directly; the memory is held by the object itself.
 
-![Shallow and Retained Size](./memory-101-images/shallow-retained.png)
+*  Implicitly, by holding references to other objects. An object holding references to other objects prevents those objects from being automatically disposed by a garbage collector (GC).
 
-### Shallow size
 
-The _shallow size_ is the size of memory that is held by the object.
+<!-- ====================================================================== -->
+## JavaScript heap and renderer memory
 
-Typical JavaScript objects have some memory reserved for their description and for storing immediate values.  Usually, only arrays and strings can have a significant shallow size.  However, strings and external arrays often have their main storage in renderer memory, exposing only a small wrapper object on the JavaScript heap.
+The JavaScript heap is a memory region in the browser process where all the JavaScript objects live. The JavaScript heap is what the **Memory** tool displays when you take a heap snapshot. The JavaScript heap is also called the _V8 heap_ (after the V8 JavaScript engine that powers Microsoft Edge).
```

Comment:
> "The JavaScript heap is what the **Memory** tool displays when you take a heap snapshot.

This used to be true, but over the last few years, upstream folks have put a lot of effort into making the V8 and Blink garbage collectors work together better. Now the heap snapshot shows both V8 memory (which contains JS and WebAssembly), and Blink objects allocated in the Blink memory region. I'm not sure how to say that in a way that is both true and easy to read.